modules/openthread/CMakeLists.txt

```diff
@@ -481,6 +486,23 @@ if(CONFIG_OPENTHREAD_SETTINGS_RAM)
   list(APPEND ot_libs openthread-platform-utils-static)
 endif()
 
+if(CONFIG_OPENTHREAD_HOSTPROCESSOR)
+  # Inject our platform spinel implementation
```

Comment:
The OpenThread build system has private compile definitions for each library. This was the easiest way to make sure exactly the same definitions are set.
Because of the C++ template for the `RadioSpinel` instance, which is header-only.